tests/testthat/test-machine_learn.R

```diff
@@ -41,3 +41,51 @@ test_that("Machine learn respects tune = FALSE", {
                       tune = FALSE, models = "xgb")
   expect_false(attr(ut, "tuned"))
 })
+
+test_that("Machine learn respects metric - when tuning (using tune_models)", {
```

Comment:
I'm unsure about the mock functions. You are right, they technically only need to test that the metric arg is getting passed through to `tune`. Tune is tested elsewhere. I think it's a little dangerous to make these kind of assumptions though, and we try to be explicit wherever possible. For example,

- What if we were to change the way metrics were passed through tune or flash? Would that break the mock function? Would this test pass when the behavior was actually not what we intended? Ambiguity is scary.
- `Tune` is tested thoroughly, and we can rely on downstream tests. Yay!
- `Flash` is not tested thoroughly. There's actually nothing in `test-flash_models` about metrics. No test  is going to fail if the metrics arg fails.
- Finally, exposing an argument that requires set values to the most basic user is risky without an error check. What happens when a user does, `machine_learn(d = pima_diabetes, outcome = diabetes, metric = NA)`, or `machine_learn(d = pima_diabetes, outcome = diabetes, metric = "percentage")`? Where is that behavior tested?

Here's what I suggest for addressing:

- Use the actual functions. `machine_learn(training_data, outcome = diabetes, metric = "PR", tune_depth = 2, n_folds = 3, models = "rf")` takes about half a second, and you'll only add 2 machine_learn calls.
- Add tests for the metric to `test-flash_models` (can copy `tune`'s tests)
- Add a test to make sure the right metric comes back for both cls and reg with machine learn's defaults.
- Add a test to verify the error/warning message when user enters NA or "garbage" to metric.

Reply:
Sounds like a plan. You are right that using mock functions could result in problems in the future. It is better to be safe. I will continue this pattern in the future. I will add these tests and one for passing the metric to `flash_models` from `machine_learn`